substrafl_examples/get_started/plot_substrafl_sklearn_fedavg.py

```diff
@@ -0,0 +1,571 @@
+from substra.sdk.schemas import AlgoSpec
+from substra.sdk.schemas import AssetKind
+
+permissions_metric = Permissions(public=False, authorized_ids=[ALGO_ORG_ID] + DATA_PROVIDER_ORGS_ID)
+
+inputs_metrics = [
+    AlgoInputSpec(
+        identifier="datasamples",
+        kind=AssetKind.data_sample,
+        optional=False,
+        multiple=True,
+    ),
+    AlgoInputSpec(identifier="opener", kind=AssetKind.data_manager, optional=False, multiple=False),
+    AlgoInputSpec(identifier="predictions", kind=AssetKind.model, optional=False, multiple=False),
+]
+
+outputs_metrics = [AlgoOutputSpec(identifier="performance", kind=AssetKind.performance, multiple=False)]
+
+objective = AlgoSpec(
+    inputs=inputs_metrics,
+    outputs=outputs_metrics,
+    name="Accuracy",
+    description=assets_directory / "metric" / "description.md",
+    file=assets_directory / "metric" / "metrics.zip",
+    permissions=permissions_metric,
+)
+
+METRICS_DOCKERFILE_FILES = [
+    assets_directory / "metric" / "iris_metrics.py",
+    assets_directory / "metric" / "Dockerfile",
+]
+
+archive_path = objective.file
+with zipfile.ZipFile(archive_path, "w") as z:
+    for filepath in METRICS_DOCKERFILE_FILES:
+        z.write(filepath, arcname=filepath.name)
+
+metric_key = clients[ALGO_ORG_ID].add_algo(objective)
+
+# %%
+# Specify the machine learning components
+# ***************************************
+#
+# In this section, you will register an algorithm and its dependencies,
+# and specify the federated learning strategy
+# as well as the nodes on which to train and to test.
+
+# %%
+# Model definition
+# ================
+#
+# The machine learning model used here is a logistic regression.
+# The `warm_start` argument is essential is this example as it indicate to use the current state of the model
+# as initialization for the future training.
+# Also, we choose to set `max_iter` to 3 to limit the `num_updates` between aggregation in order to maximize
+# the communication between organizations.
+
+from sklearn import linear_model
+
+cls = linear_model.LogisticRegression(random_state=SEED, warm_start=True, max_iter=3)
+
+
+# %%
+# Substrafl algo definition
+# ==========================
+#
+# This section is the most important one for this example. We will define here the function that will run locally on
+# each node to train the model using sklearn function.
+#
+# As SubstraFL does not provide an SklearnFedAvgAlgo, we need to define one using the provided documentation on
+# :ref:`substrafl_doc/api/algorithms:Base Class`
+#
+# To define a custom algorithm, we will need to inherit from the base class Algo, and to define two properties and four
+# functions:
+#
+# - **strategies** (property): the list of strategy name our algorithm is compatible with.
+# - **model** (property): a property that return the model from the defined algo.
+# - **train** (function): a function with the @remote_data decorator. This function describe the training process to
```

Comment:
You did explain well why we need a decorator in the function docstring, so I would remove it from here. (same for predict).